Orange/data/table/base.py

```diff
@@ -0,0 +1,1409 @@
+              row variable and the j-th value of the column variable.
+            - for continuous variables, the contingency is a list of two arrays,
+              where the first array contains ordered distinct values of the
+              column_variable and the element (i, j) of the second array
+              contains count of rows with i-th value of the row variable
+              and j-th value of the ordered column variable.
+
+            The final output structure looks like:
+            result (tuple)
+             |__contingencies (list of len(col_vars))
+             |   |__contingency, one of
+             |   |   |__2D np.array
+             |   |   |__tuple(sorted list of continuous values, 2D np.array)
+             |   |__1D np.array (missing column elements for each row value)
+             |__num missing row_var values (int)
+        """
+        if row_var is None:
+            row_var = self.domain.class_var
+            if row_var is None:
+                raise ValueError("No row variable. ")
+        if col_vars is None:
+            col_vars = self.domain.attributes
+
+        row_var = self.domain[row_var]
+        col_vars = [self.domain[v] for v in col_vars]
+        if not row_var.is_discrete:
+            raise TypeError("Row variable must be discrete. ")
+        if any(not (var.is_discrete or var.is_continuous) for var in col_vars):
+            raise ValueError("Contingency can be computed only for discrete and continuous values. ")
+
+        # assertions at this point:
+        #  - row_var exists and is discrete
+        #  - col_vars is a list of continuous or discrete variables
+        contingencies = []
+        unknown_grouper = self.groupby(row_var)
+        for var in col_vars:
+            if var is row_var:
+                # we can't do a pivot table of a variable with itself easily,
+                # so instead just compute the distributions, which are equivalent
+                # row_var is always discrete
+                dist_unks = self._compute_distributions(columns=[row_var])
+                contingencies.append(dist_unks[0])
+            else:
+                pivot = pd.pivot_table(self, values=self._WEIGHTS_COLUMN, index=[row_var],
+                                       columns=[var], aggfunc=np.sum).fillna(0)
+                unknowns = unknown_grouper[var].agg(lambda x: x.isnull().sum())
+                if var.is_discrete:
+                    contingencies.append((pivot.values, unknowns.values))
+                else:
+                    contingencies.append(((pivot.columns.values, pivot.values), unknowns.values))
+        return contingencies, self[row_var].isnull().sum()
+
+    @property
+    def density(self):
+        """Return the density of the current table."""
+        raise NotImplementedError
+
+    @property
+    def is_sparse(self):
+        """Return True if the current table is sparse."""
+        raise NotImplementedError
```

Comment:
I mean, what does this even mean?

> Return True if the current table is sparse

What are the implications of this? If I extend `TableBase` into a custom type that uses list or dict or file as a storage, what do I have it return here?